### DISCUSSION TITLE
$doctrine->persist() not equal for mariadb and prostgresql

### DISCUSSION THREAD

**Discussion by ghost** [symfony/symfony#52340] [Q&A] [answered]
Hi all,

My problem is construct collection with one flush eg : 
```
$user = new user();
$user->setName('toto');
$doctrine->persist($user);

$store = new store();
$store->setUser($user);
$doctrine->persist($store)

dd($user->getId());

$doctrine->flush;
```
under postgresql $user->getId() show 1
under mariadb $user->getId() show null

How to for mariadb persist too ? 
Thx for your response :)

**Comment by stof** [comment, marked as answer]
Auto-incremented ids for MySQL or MariaDB are available only after flush (when it is saved in the DB).
For Postgresql, the ORM uses a sequence to generate ids, so after persisting, the id is already available (generating the id is decoupled from saving)

**Reply by stof** [reply]
The guarantee done by Doctrine is that the generated ID is available after the flush. It may be available before but that's a "it depends".

**Comment by ghost** [comment]
Hi,

Thank you for your response, after posting, I searched the web and did tests in the meantime, that's even why I closed the discussion. Mariadb very recently put the sequence system and still after some tests, the work is not identical.
As you said, PostgreSQL generates it on the fly because it's in its DNA.
Thanks again and have a good week :)
